File: PYTHON/EJERCICIOS/Ejercicio20-entrada-compleja.py
# pedimos al usuario la opcion:
opcion = int(input("""INGRESE UNA OPCION:
                   1. SUMAR
                   2. DIVIDIR
                   3. MULTIPLICAR
                   4. RESTAR: """))
numero_01 = int(input("INGRESA UN NUMERO: "))
numero_02 = int(input("INGRESA UN NUMERO: "))
# evaluamos
if opcion == 1:
    print("RESULTADO: ",(numero_01 + numero_02))
elif opcion == 2:
    # validamos la division
    if numero_02 == 0:
        print("No se puede dividir por 0.")
    else:
        print("RESULTADO: ",(numero_01 / numero_02))
elif opcion == 3:
        print("RESULTADO: ",(numero_01 * numero_02))
elif opcion == 4:
        print("RESULTADO: ",(numero_01 - numero_02))
else:
    print("INGRESE UNA OPCION VALIDA.")

    
    
        
    
    

File: JAVA/EJERCICIOS/CONDICIONALES/Ejercicio20EntradaCompleja.java
package JAVA.EJERCICIOS.CONDICIONALES;
/*
EJERCICIO DE LOGICA DE PROGRAMACION: 
crear un programa que funcione como una calculadora
básica interactiva, permitiendo al usuario seleccionar
entre sumar, restar, multiplicar o dividir dos números
ingresados, y mostrando el resultado o mensajes 
de error en caso de entradas inválidas o divisiones por cero.
 */
import java.util.Scanner;
public class Ejercicio20EntradaCompleja {
    static Scanner leer = new Scanner(System.in);
    public static void main(String[] args) {
        // variables
        int opcion, n1, n2;
        // pedimos los datos
        System.out.print("Ingresa una opcion: " +
                "\n1. Sumar" +
                "\n2. Dividir" +
                "\n3. Multiplicar" +
                "\n4. Restar: ");
        opcion = leer.nextInt();
        System.out.print("Ingresa el primer numero: ");
        n1 = leer.nextInt();
        System.out.print("Ingresa el segundo numero: ");
        n2 = leer.nextInt();
        // evaluamos
        switch (opcion) {
            case 1:
                System.out.println("RESULTADO: " + (n1 + n2));
                break;
            case 2:
                // validamos la division
                if (n2 == 0) {
                    System.out.println("No se puede dividir por cero.");
                } else {
                    System.out.println("RESULTADO: " + (n1 / n2));
                }
                break;
            case 3:
                System.out.println("RESULTADO: " + (n1 * n2));
                break;
            case 4:
                System.out.println("RESULTADO: " + (n1 - n2));
                break;
            default:
                System.out.println("Ingrese una opcion valida!.");
                break;
        }
    }
}
